interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -62,16 +62,20 @@ type PromptConstraints struct {
 	Path                 string   `json:"path"`
 	Permissions          []string `json:"permissions"`
 	AvailablePermissions []string `json:"available-permissions"`
+	// Preserve originally-requested permissions so we can send a response with
+	// explicitly-allowed/denied permissions, even if some permissions were
+	// previously satisfied.
+	originalPermissions []string
```

Comment:
looking at current code, when a new rule is added which may subtract permissions, shouldn't those also be subtracted from `origianlPermissions` ?

Reply:
No, the idea here is that satisfied permissions should be subtracted from the `Permissions` list, but should be preserved in the `originalPermission` list. The former is what we present to the user (just the unsatisfied permissions), while the latter is what we send back to the kernel in the end when everything has been satisfied.